test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/SaveTempDataFilterTest.cs

```diff
@@ -28,31 +30,73 @@ public static TheoryData<IActionResult> ActionResultsData
         }
 
         [Fact]
-        public void OnResultExecuting_RegistersOnStartingCallback()
+        public async Task OnResultExecuting_DoesntThrowIfResponseStarted()
+        {
+            // Arrange
+            var responseFeature = new TestResponseFeature(hasStarted: true);
+            var httpContext = GetHttpContext(responseFeature);
+            var tempDataFactory = new Mock<ITempDataDictionaryFactory>(MockBehavior.Loose);
+            tempDataFactory
+                .Setup(f => f.GetTempData(It.IsAny<HttpContext>()))
+                .Verifiable();
+            var filter = new SaveTempDataFilter(tempDataFactory.Object);
+            var context = GetResultExecutingContext(httpContext);
+            filter.OnResultExecuting(context);
+
+            // Act
+            await responseFeature.FireOnSendingHeadersAsync();
```

Comment:
Is it possible to use the same technique to make a functional test that does this?